mod-pubsub-server/src/test/java/org/folio/services/impl/SecurityManagerTest.java

```diff
@@ -199,6 +203,30 @@ public void shouldNotCreatePubSubUserIfItExists(TestContext context) {
     }).onComplete(context.asyncAssertSuccess());
   }
 
+  @Test
+  public void shouldNotCreatePubSubUserIfEnvVariable(TestContext context) {
+    String userId = UUID.randomUUID().toString();
+    String userCollection = new JsonObject()
+      .put("users", new JsonArray().add(existingUpToDateUser(userId)))
+      .put("totalRecords", 1).encode();
+
+    stubFor(get(USERS_URL_WITH_QUERY).willReturn(ok().withBody(userCollection)));
+
+    OkapiConnectionParams params = new OkapiConnectionParams(headers, vertx);
+
+    Future<Void> future = securityManagerNoSystemUser.createPubSubUser(params);
+
+    future.map(ar -> {
+      List<LoggedRequest> requests = findAll(RequestPatternBuilder.allRequests());
+      assertEquals(0, requests.size());
+
+      // Verify user create request has not sent
+      verify(0, new RequestPatternBuilder(POST, urlEqualTo(USERS_URL)));
+
+      return null;
+    }).onComplete(context.asyncAssertSuccess());
```

Comment:
```suggestion
    securityManagerNoSystemUser.createPubSubUser(params)
    .onSuccess(ar -> {
      List<LoggedRequest> requests = findAll(RequestPatternBuilder.allRequests());
      assertEquals(0, requests.size());

      // Verify that user create request has not been sent
      verify(0, new RequestPatternBuilder(POST, urlEqualTo(USERS_URL)));
    }).onComplete(context.asyncAssertSuccess());
```

- inlined variable `future`
- used `onSuccess()` instead of `map()` in order to avoid returning a null result
- fixed comment